cmake/tpls/DyninstBoost.cmake

```diff
@@ -17,33 +17,27 @@
 
 include_guard(GLOBAL)
 
-# Need at least 1.71 for a usable BoostConfig.cmake
-set(_min_version 1.71.0)
+# Newer CMake prefer using the BoostConfig.cmake shipped with Boost
+if(POLICY CMP0167)
+  cmake_policy(SET CMP0167 NEW)
+endif()
 
 # Use multithreaded libraries
 set(Boost_USE_MULTITHREADED ON)
 
 # Don't use libraries linked statically to the C++ runtime
 set(Boost_USE_STATIC_RUNTIME OFF)
 
-if(Boost_ROOT_DIR)
```

Comment:
Because CMake's [search rules](https://cmake.org/cmake/help/latest/command/find_package.html#config-mode-search-procedure) are complex, all of our custom finders use the convention of `Package_ROOT_DIR` to force CMake to use the given location. Disabling searching system paths is necessary (but not necessarily sufficient) to make that happen. Translating from Boost_ROOT_DIR to Boost_ROOT also alleviates needing an extra HINT.

Reply:
(IMHO using such a convention is evil and confusing. I understand the desire to assert e.g. "Boost comes from this directory" but muddying CMake's already heady conventions with your own is an abominable approach. But not my project nor a feature I require, so I'll be quiet now.)

The latest version of the patch has the intended behavior, either `-DBoost_ROOT_DIR` or `-DBoost_ROOT` will find an externally-installed Boost and take precedence over a system-installed Boost.

Reply:
>(IMHO using such a convention is evil and confusing. I understand the desire to assert e.g. "Boost comes from this directory" but muddying CMake's already heady conventions with your own is an abominable approach. But not my project nor a feature I require, so I'll be quiet now.)

Worry not, I share the sentiment. But this addresses two of my biggest gripes with CMake: a lack of configurability and a lack of "standard" conventions. I spent an unreasonable amount of time researching this and took up yet another unreasonable amount of Jim's time deciding on what we really wanted to do. But "use only this directory" was most important to us. Given what I've learned from CMake talks in the interim, I would have just added a dyninst-specific flag to constrain the search process instead of using different _ROOT specifiers.

>The latest version of the patch has the intended behavior: either -DBoost_ROOT_DIR or -DBoost_ROOT will find an externally-installed Boost, the former will fail if not found and the latter can fall back to a system-installed Boost.

Well, I *was* going to say that's always been the case by design, but I just noticed that `Boost_ROOT_DIR` didn't properly fail when pointed to a bad Boost install without this PR. I'm fairly certain I had tested that while I was working on the CMake updates. Ugh. I'll add more tests.

```
$ cmake .. -DBoost_ROOT=/home/tim/Downloads/boost_1_86_0/install
                                                  ^^^^^
...
Found Boost 1.86.0
Boost include directories: /home/tim/Downloads/boost_1_86_0/install/include

$ cmake .. -DBoost_ROOT=/home/tim/Downloads/boost_1_66_0/install
                                                  ^^^^^
...
Found Boost 1.74.0
-- Boost include directories: /usr/include
```